src/main/java/org/schemaspy/mutation/Mutation.java

```diff
@@ -0,0 +1,19 @@
+package org.schemaspy.mutation;
+
+import org.schemaspy.model.Table;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Abstraction for mutators that will modify the model.
+ * Model since a table has access to the whole object graph.
+ */
+public interface Mutation<T> {
+    /**
+     * Perform mutation on tables
+     * @param tables that should be mutated
+     * @return list of mutations
+     */
+    List<T> mutate(Collection<Table> tables);
```

Comment:
Remove parameter. A better design is to let implementors encapsulate the tables if they are needed.

Reply:
https://github.com/schemaspy/schemaspy/pull/921#discussion_r896912261
> Separate creation and execution

https://github.com/schemaspy/schemaspy/pull/921#discussion_r896915958
> `ImpliedConstraintsFinder` should not take a boolean. Instead, the command line argument should determine what your object graph looks like.

Since there is no object graph over how the application should work and I assume we want to create the graph in Main in reality, so there is no possibility of having Table as a constructor argument.

The closes to this is if in Entry point. Produce something in the line of 
```
new SchemaAnalyzer(
  new DbWithImpliedConstraints(
    new DbWithRailsConstraints(
      new DbWithXmlAugmentation(
        new DatabaseFrom(
          new JDBCSource(connectionProperties)
         )
      )
   ),
   new XmlMedium(/*.....*/),
   new JsonMedium(/*.....*/),
   new HtmlMedium(/*.....*/)
).analyze();
```

Which probably would make more sense, to have constructor with something that can produce a Database or something that can take a Medium,

Reply:
> I assume we want to create the graph in Main in reality, so there is no possibility of having Table as a constructor argument.

The design you propose (which I agree with) does this so I'm not sure what point you're trying to make. `DbWithImpliedConstraints` would have access to the decoratee who can be asked for `Tables` (or whatever contract we end up with when we get there).

Reply:
I'm only disagreeing with tables as an encapsulatee and it relates to the other comments. That a good encapsulatee doesn't exist at the moment and as such this isn't the implementation we want.